libres/lib/enkf/hook_manager.cpp

```diff
@@ -228,3 +222,51 @@ hook_manager_iget_hook_workflow(const hook_manager_type *hook_manager,
 int hook_manager_get_size(const hook_manager_type *hook_manager) {
     return vector_get_size(hook_manager->hook_workflow_list);
 }
+
+bool Runpath::operator<(const Runpath &other) const {
+    if (iter < other.iter)
+        return true;
+    if (iens < other.iens)
+        return true;
+    return false;
+}
+
+/**
+ * @param iens The ensamble index of the run
+ * @param iter The iteration of the run
+ * @param runpath The absolute path to the run,
+ *     ie. <...>/poly_example/poly_out/realization-0/iter-0
+ * @param jobname The name of the job
+ */
+Runpath::Runpath(int iens, int iter, std::string runpath, std::string jobname)
+    : iens(iens), iter(iter), runpath(runpath), jobname(jobname) {}
+
+Runpath::Runpath(int iens, int iter, const char *runpath_, const char *jobname_)
+    : iens(iens), iter(iter) {
+    if (jobname_ == NULL) {
+        jobname = "";
+    } else {
+        jobname = std::string(jobname_);
+    }
+
+    if (runpath_ == NULL) {
+        runpath = "";
+    } else {
+        runpath = std::string(runpath_);
+    }
+}
+
+/** Writes the runpath_list_file
+ *
+ * @param runpath_list The list of Runpath to go into the file.
+ */
+void hook_manager_write_runpath_file(const hook_manager_type *hook_manager,
+                                     std::vector<Runpath> runpath_list) {
+    auto stream = mkdir_fopen(fs::path(hook_manager->runpath_list_file), "w");
+    std::sort(runpath_list.begin(), runpath_list.end());
+    for (Runpath rp : runpath_list) {
+        fprintf(stream, "%03d  %s  %s  %03d\n", rp.iens, rp.runpath.c_str(),
+                rp.jobname.c_str(), rp.iter);
```

Comment:
For new `fprintf` use `#include <fmt/printf.h>`
```suggestion
        fmt::fprintf(stream, "%03d  %s  %s  %03d\n", rp.iens, rp.runpath,
                rp.jobname, rp.iter);
```
or preferable the main fmt API:
```suggestion
        fmt::print(stream, "{:03d}  {}  {}  {:03d}\n", rp.iens, rp.runpath,
                rp.jobname, rp.iter);
```

Reply:
Fixed it, mind I believe we are pretty sensitive to the file contents remaining exactly the same